protocol/contracts/v2/strategies/RamosStrategy.sol

```diff
@@ -0,0 +1,186 @@
+            // https://docs.balancer.fi/reference/lp-tokens/underlying.html#overview
+            // https://docs.balancer.fi/concepts/advanced/valuing-bpt.html#on-chain
+            uint256 bptTotalSupply = IBPT(address(bptToken)).getActualSupply();
+            if (bptTotalSupply > 0) {
+                uint256 bptBalanceInRamos = amoStaking.stakedBalance() + bptToken.balanceOf(address(amoStaking));
+                (, uint256 stableBalanceInLP) = poolHelper.getTempleStableBalances();
+
+                stableBalanceInRamos = stableBalanceInLP * bptBalanceInRamos / bptTotalSupply;
+            }
+        }
+
+        // get the RAMOS strategy assets
+        uint256 length = assets.length;
+        assetBalances = new AssetBalance[](length);
+
+        address asset;
+        uint256 ramosStrategyBalance;
+        for (uint256 i; i < length; ++i) {
+            asset = assets[i];
+
+            if (asset == stable) {
+                // Sum the RAMOS balance and this contract's balance of the ERC20.
+                ramosStrategyBalance = stableBalanceInRamos + IERC20(asset).balanceOf(address(this));
+            } else {
+                // Sum the `amoStaking` balance, `rewardsRecipient` balance and this contract's balance of the ERC20.
+                // But since `rewardsRecipient` address may be a hot wallet with a bunch of different things going on, we won't include that balance
+                ramosStrategyBalance = IERC20(asset).balanceOf(address(amoStaking)) + IERC20(asset).balanceOf(address(this));
+            }
+
+            assetBalances[i] = AssetBalance({
+                asset: asset,
+                balance: addManualAssetBalanceDelta(ramosStrategyBalance, asset)
+            });
+        }
+
+        debt = currentDebt();
+    }
+
+    /**
+     * @notice Borrow a fixed amount from the Treasury Reserves and add liquidity
+     * These stables are sent to the RAMOS to add liquidity and stake BPT
+     */
+    function borrowAndAddLiquidity(uint256 _amount, AMO__IBalancerVault.JoinPoolRequest memory _requestData) external onlyElevatedAccess {
+        // Borrow the DAI and send it to RAMOS. This will also mint `dUSD` debt.
+        treasuryReservesVault.borrow(_amount, address(ramos));
+        emit BorrowAndAddLiquidity(_amount);
+
+        // Add liquidity
+        ramos.addLiquidity(_requestData);
+    }
+
+    /**
+     * @notice Remove liquidity and repay debt back to the Treasury Reserves
+     * The stables from the removed liquidity are sent from RAMOS to this address to repay debt
+     */
+    function removeLiquidityAndRepay(AMO__IBalancerVault.ExitPoolRequest memory _requestData, uint256 _bptAmount) external onlyElevatedAccess {
+        // Remove liquidity
+        ramos.removeLiquidity(_requestData, _bptAmount, address(this));
+
+        // Repay debt back to the Treasury Reserves
+        IERC20 stable = treasuryReservesVault.stableToken();
```

Comment:
There's already an immutable variable you can use, inherited from `AbstractStrategy`

`stableToken`